qa/rpc-tests/bumpfee.py

```diff
@@ -0,0 +1,198 @@
+#!/usr/bin/env python3
+# Copyright (c) 2016 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework import blocktools
+from test_framework.mininode import CTransaction
+from test_framework.util import *
+from test_framework.util import *
+
+import io
+import time
+
+class BumpFeeTest (BitcoinTestFramework):
+
+    def __init__(self):
+        super().__init__()
+        self.num_nodes = 2
+        self.setup_clean_chain = True
+
+    def setup_network(self, split=False):
+        self.nodes = []
+        self.nodes.append(start_node(0, self.options.tmpdir, ["-debug", "-walletrbf"]))
+        self.nodes.append(start_node(1, self.options.tmpdir, ["-debug"]))
+        connect_nodes_bi(self.nodes,0,1)
+        self.is_network_split=False
+        self.sync_all()
+
+    def create_fund_sign_send(self, node, outputs, feerate=0):
+        if feerate != 0:
+            node.settxfee(feerate)
+        rawtx = node.createrawtransaction([], outputs)
+        fundtx = node.fundrawtransaction(rawtx)
+        signedtx = node.signrawtransaction(fundtx['hex'])
+        txid = node.sendrawtransaction(signedtx['hex'])
+        return txid
+
+
+    def run_test (self):
+
+        print("Mining blocks...")
+
+        # fund node0 with 10 coins of 0.001 btc (100,000 satoshis)
+        self.nodes[1].generate(110)
+        self.sync_all()
+        a0 = self.nodes[0].getnewaddress()
+        a1 = self.nodes[1].getnewaddress()
+        for i in range(10):
+            self.nodes[1].sendtoaddress(a0, 0.001)
+        self.sync_all()
+        self.nodes[1].generate(1)
+        self.sync_all()
+        assert_equal(self.nodes[0].getbalance(), Decimal('0.01'))
+
+        # create and bump an RBF transaction
```

Comment:
This test is long and it's not clear right away whether the different parts of it depend on each other. If it could be broken up in some way, I think it would be more readable and easier to work with in the future. E.g.

```
self.test_simple_bumpfee_succeeds(a1)
self.test_nonrbf_bumpfee_fails(a1)
self.test_notmine_bumpfee_fails(a0)
...
```